src/main/java/de/cyklon/reflection/entities/OfflinePackage.java

```diff
@@ -0,0 +1,59 @@
+package de.cyklon.reflection.entities;
+
+import de.cyklon.reflection.entities.impl.OfflinePackageImpl;
+import de.cyklon.reflection.function.Filter;
+import de.cyklon.reflection.types.Loadable;
+import de.cyklon.reflection.types.Nameable;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.Unmodifiable;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public interface OfflinePackage extends Nameable, Loadable<ReflectPackage> {
```

Comment:
Add getClass, getOptionalClass, getPackage and getOptionalPackage. These methods should be implemented in the interface itself via default methods. The methods for classes already exist in ReflectPackage and should be moved in OfflinePackage, getPackage and getOptionalPackage has to be created but can be done similar to the implementation for class